### PR TITLE
Add experiments to `VersionResponse`

### PR DESCRIPTION
**Pull request by domesticmouse**
fixes: https://github.com/dart-lang/dart-pad/issues/2429

### CODE REVIEW

File: lib/src/common_server_impl.dart [hunk 1]
```diff
@@ -274,7 +274,14 @@ class CommonServerImpl {
         ..flutterDartVersionFull = _sdk.versionFull
         ..flutterVersion = _sdk.flutterVersion
         ..packageVersions.addAll(packageVersions)
-        ..packageInfo.addAll(packageInfos),
+        ..packageInfo.addAll(packageInfos)
+        ..experiment.addAll(
+          [
+            // TODO(DomesticMouse): update when records and patterns experiments
+            // are modified or removed.
+            if (_sdk.masterChannel) ...['records', 'patterns'],
```
Comment (johnpryan) [inline lib/src/common_server_impl.dart:282]: Do we set these manually when we deploy? Is there a single source of truth somewhere?

Comment (domesticmouse) [reply lib/src/common_server_impl.dart:282]: https://github.com/dart-lang/dart-services/pull/972/files

Comment (johnpryan) [reply lib/src/common_server_impl.dart:282]: One nitpick would be to keep a top-level list of experiments to avoid repeating ourselves, otherwise LGTM

**Review by johnpryan — APPROVED** [review]
LGTM - thanks for the quick turnaround. Feel free to land and I'll update the frontend

**Pull request merged**